### DISCUSSION TITLE
API create file PUT gives Problems parsing JSON error

### DISCUSSION THREAD

**Discussion by NetsuiteVS** [community/community#21447] [Other Feature Feedback, Questions, & Ideas]
<p>I try to create file from a Netsuite (more or less javascript) script file. When I try to create a file I get “Problems parsing JSON” error message. I tried it from Postman and it gives me the same error when tring to set Body as “form-data” but it was success when set Body as “Raw”. (on one picture because new user can upload only one…)<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/4/7/47d466e9b115329bba7324a3d0339698fd52a846.png" data-download-href="/uploads/short-url/afqY77I4PA0c4PIyiVtPFljqGpg.png?dl=1" title="Screenshot_1" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/87364860/181064592-5b184821-7de7-43a3-9d35-09f7d177072c.png" alt="Screenshot_1" data-base62-sha1="afqY77I4PA0c4PIyiVtPFljqGpg" width="537" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/4/7/47d466e9b115329bba7324a3d0339698fd52a846_2_537x500.png, https://aws1.discourse-cdn.com/github/optimized/3X/4/7/47d466e9b115329bba7324a3d0339698fd52a846_2_805x750.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/4/7/47d466e9b115329bba7324a3d0339698fd52a846.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/4/7/47d466e9b115329bba7324a3d0339698fd52a846_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot_1</span><span class="informations">965×898 30.5 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Code:</p>
<pre><code class="lang-auto">var url = 'https://api.github.com/repos/xxx/yyy/contents/new5.js';

var headersObj = {
    "Content-Type": "application/vnd.github.v3+json",
    "Authorization": "token &lt;valid token&gt;"
};

var bodyObj = {
    "message": "commit message",
    "content": fileObjBase64
};

var apiResponse = https.put({
    url: url,
    headers: headersObj,
    body: bodyObj
});
</code></pre>
<p>This gives me “Problems parsing JSON” error too.<br>
I need some help, how to set Body to raw or any other idea to create file.</p>

**Comment by tajmone** [comment]
<aside class="quote no-group" data-username="NetsuiteVS" data-post="1" data-topic="190218">
<div class="title">
<div class="quote-controls"></div>
 NetsuiteVS:</div>
<blockquote>
<p>I tried it from Postman and it gives me the same error when tring to set Body as “form-data” but it was success when set Body as “Raw”</p>
</blockquote>
</aside>
<p>This might be related to the raw data being passed in base64 encoding, and the need to escape the encoded string in one context. See:</p>
<aside class="onebox stackexchange" data-onebox-src="https://stackoverflow.com/questions/51679672/how-to-format-base64-encoded-string-while-sending-in-json">
  <header class="source">

      <a href="https://stackoverflow.com/questions/51679672/how-to-format-base64-encoded-string-while-sending-in-json" target="_blank" rel="noopener nofollow ugc">stackoverflow.com</a>
  </header>

  <article class="onebox-body">
      <a href="https://stackoverflow.com/users/9829517/learn-groovy" target="_blank" rel="noopener nofollow ugc">
    <img alt="learn groovy" src="https://user-images.githubusercontent.com/11064292/181064594-3fab6897-e899-4376-81a4-ca005f414b34.png" class="thumbnail onebox-avatar" width="128" height="128">
  </a>

<h4>
  <a href="https://stackoverflow.com/questions/51679672/how-to-format-base64-encoded-string-while-sending-in-json" target="_blank" rel="noopener nofollow ugc">How to format Base64 encoded string while sending in JSON</a>
</h4>

<div class="tags">
  <strong>javascript, java, groovy, base64</strong>
</div>

<div class="date">
  asked by
  
  <a href="https://stackoverflow.com/users/9829517/learn-groovy" target="_blank" rel="noopener nofollow ugc">
    learn groovy
  </a>
  on <a href="https://stackoverflow.com/questions/51679672/how-to-format-base64-encoded-string-while-sending-in-json" target="_blank" rel="noopener nofollow ugc">08:25PM - 03 Aug 18 UTC</a>
</div>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

**Comment by NetsuiteVS** [comment]
<p>Thank you for your response. As you can see on the picture the base64 text doesn’t contains any special character except one =. Althrough I tried to remove  ‘=’ char and I still get the same error.</p>

**Comment by tajmone** [comment]
<p>In the above examples, the contents of <code>fileObjBase64</code> are not visible. I believe that base64 encoded contents need some field specifying that it’s encoded data. Eg:</p>
<pre><code class="lang-auto">{
    "data": "JVBERi0xLjMNJeLjz9MNCjc ...
</code></pre>
<p>or something like that. In <a href="https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/Data_URIs" rel="noopener nofollow ugc">DataURIs</a> you also have to add some prefix to let the parser know it’s encoded data.</p>
<p>There must be something missing there which doesn’t allow the JSON parser to realize that some base64 data is coming — the whole purpose of Base64 encoding is, after all, to convert binary data (which might contain invalid bytes for a string) into a parseable string of ASCII chars.</p>

**Comment by NetsuiteVS** [comment, marked as answer]
<p>You can see the base64 content on the picture. In the content parameter’s value.<br>
Meanwhile I found the solution. Netsuite transformed the body json in the background from<br>
“message”: “some message”<br>
to<br>
message": {“some message”} .<br>
Solution is giving json as string format:</p>
<pre><code class="lang-auto">var bodyObj = JSON.stringify({
    "message": "commit message",
    "content": fileObjBase64
});
</code></pre>
